src/main/java/com/api/YouTubeParser.java

```
@@ -0,0 +1,70 @@
+        try {
+            credential = Authorization.authorize(scopes, "commentThreads");
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+        }
+        youtube = new YouTube.Builder(Authorization.HTTP_TRANSPORT, Authorization.JSON_FACTORY, credential)
+                .setApplicationName("youtube comments parser").build();
+    }
+
+    public void getCommentsFromVideos(List<String> videoId) {
+        for (String video : videoId) {
+            List<CommentThread> videoComments = getCommentThreads(video);
+            CommentSnippet snippet;
+            YouTubeVideo youtubeVideo = new YouTubeVideo();
+
+            for (CommentThread videoComment : videoComments) {
+                snippet = videoComment.getSnippet().getTopLevelComment().getSnippet();
+                Comment comment = new Comment.Bilder()
+                        .setAuthorName(snippet.getAuthorDisplayName())
+                        .setMessageText(snippet.getTextDisplay())
+                        .setLike(snippet.getLikeCount())
+                        .setLastModified(snippet.getPublishedAt())
+                        .setEditing(!(snippet.getUpdatedAt().equals(snippet.getPublishedAt())))
+                        .build();
+                youtubeVideo.getComments().add(comment);
+            }
+            listOfVideos.add(youtubeVideo);
+        }
+    }
+
+    private static List<CommentThread> getCommentThreads(String videoId) {
+        CommentThreadListResponse videoCommentsListResponse = null;
+        try {
+            videoCommentsListResponse = youtube.commentThreads().list("snippet")
+                    .setVideoId(videoId).setTextFormat("plainText").execute();
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage());
+        }
+        if (videoCommentsListResponse != null) {
+            return videoCommentsListResponse.getItems();
+        }
+        throw new NullPointerException("NullPointerException");
```

It is better to throw a custom exception with a reasonable description.